IPython/core/interactiveshell.py

```diff
@@ -208,6 +208,16 @@ class ExecutionResult(object):
     def success(self):
         return (self.error_before_exec is None) and (self.error_in_exec is None)
 
+    def raise_error(self):
+        """Reraises error if `success` is `True`, otherwise does nothing"""
+        if not self.success:
+            return
```

Comment:
In fact, this check is redundant - since success just means both errors are None, the two `if` lines below are doing the same check.